app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeavePresenter.java

```diff
@@ -72,98 +71,128 @@ public void onRequestError(BaseException error) {
 
     @Override
     public boolean validateDataInput(Request request) {
-        validateProjectName(request.getProject());
         switch (request.getLeaveType().getName()) {
             case Constant.LeaveType.IN_LATE_A:
             case Constant.LeaveType.IN_LATE_M:
-                validateCheckinTime(request.getCheckinTime());
-                validateCompensationFrom(request.getCompensation().getFromTime());
-                validateCompensationTo(request.getCompensation().getToTime());
-                validateReason(request.getReason());
+                if (validateCheckinTime(request.getCheckinTime())
+                        & validateCompensationFrom(request.getCompensation().getFromTime())
+                        & validateCompensationTo(request.getCompensation().getToTime())
+                        & validateReason(request.getReason())
+                        & validateProjectName(request.getProject())) {
+                    return true;
+                }
                 break;
             case Constant.LeaveType.LEAVE_EARLY_A:
             case Constant.LeaveType.LEAVE_EARLY_M:
-                validateCheckoutTime(request.getCheckoutTime());
-                validateCompensationFrom(request.getCompensation().getFromTime());
-                validateCompensationTo(request.getCompensation().getToTime());
-                validateReason(request.getReason());
+                if (validateCheckoutTime(request.getCheckoutTime())
+                        & validateCompensationFrom(request.getCompensation().getFromTime())
+                        & validateCompensationTo(request.getCompensation().getToTime())
+                        & validateReason(request.getReason())
+                        & validateProjectName(request.getProject())) {
```

Comment:
ERROR: [STYLE][NS_DANGEROUS_NON_SHORT_CIRCUIT] Potentially dangerous use of non-short-circuit logic in com.framgia.wsm.screen.requestleave.RequestLeavePresenter.validateDataInput(Request)
http://findbugs.sourceforge.net/bugDescriptions.html#NS_DANGEROUS_NON_SHORT_CIRCUIT